source/MaterialXGenGlsl/GlslShaderGenerator.cpp

```diff
@@ -454,16 +458,17 @@ void GlslShaderGenerator::emitPixelStage(const ShaderGraph& graph, GenContext& c
     }
 
     // Emit uv transform function
+    StringMap transformMap;
     if (context.getOptions().fileTextureVerticalFlip)
     {
-        emitInclude("stdlib/" + GlslShaderGenerator::LANGUAGE + "/lib/mx_get_target_uv_vflip.glsl", context, stage);
-        emitLineBreak(stage);
+        transformMap["$flip"] = "1.0 - ";
     }
     else
     {
-        emitInclude("stdlib/" + GlslShaderGenerator::LANGUAGE + "/lib/mx_get_target_uv_noop.glsl", context, stage);
-        emitLineBreak(stage);
+        transformMap["$flip"] = "";
     }
+    emitInclude("stdlib/" + GlslShaderGenerator::LANGUAGE + "/lib/mx_transform_uv.glsl", context, stage, &transformMap);
```

Comment:
I don't think you need to handle token substitution explicitly for included content. Substitution is done on the final source code for each stage, so if you just set the flip token in the generators's substitution map (`_tokenSubstitutions`) this will be substituted automatically in that final substitution pass.

Reply:
Yes. This makes sense. I'll add a better token name and put it there.